src/MySqlConnector/MySqlDecimal.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace MySqlConnector;
+
+public readonly struct MySqlDecimal
+{
+	private readonly string value;
+
+	internal MySqlDecimal(string val)
+	{
+		string regexWithDecimal = @"^-?([1-9][0-9]*|0)(\.[0-9]+)?$";
+		string regexWithOutDecimal = @"^-?([1-9][0-9]*|0)$";
+		var matchWithDecimal = Regex.Match(val, regexWithDecimal, RegexOptions.IgnoreCase);
+		var matchWithOutDecimal = Regex.Match(val, regexWithOutDecimal, RegexOptions.IgnoreCase);
+
+		if(!(matchWithDecimal.Success || matchWithOutDecimal.Success))
+		{
+			throw new MySqlConversionException("Format is wrong.");
```

Comment:
Use `FormatException`. (`MySqlConversionException` was for backwards compatibility for MySql.Data and is documented as "thrown when a MySQL value can't be converted to another type", which isn't true here: we're trying to construct a MySQL value.)